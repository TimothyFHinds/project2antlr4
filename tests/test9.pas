program test9;

var 
A : REAL;
B : REAL = 12.0;

procedure ScopeInner(var c : real);
var A : REAL;
begin
    A := 10;
    //writeln (A)
end;

begin (* Main *)
    writeln('Old Value of A');
    A := 20;
    writeln (A);
    ScopeInner(B);
    writeln('New Value of A');
    writeln (A);
end. 
(* Main *)

