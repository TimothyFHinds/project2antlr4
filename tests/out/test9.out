'Old Value of A'
20.0
ALREADY A GLOBAL VARIABLE!
'New Value of A'
10.0
